streams/cluster/src/main/java/com/hortonworks/streamline/streams/cluster/service/metadata/StormMetadataService.java

```diff
@@ -153,45 +177,49 @@ private Client newHttpClient() {
-
-            if (stormUiComp == null) {
-                throw new ServiceComponentNotFoundException(clusterId, ServiceConfigurations.STORM.name(), ComponentPropertyPattern.STORM_UI_SERVER.name());
-            }
-
+            final Component stormUiComp =
+                    EnvironmentServiceUtil.getComponent(
+                            environmentService, clusterId, AMBARI_JSON_SERVICE_STORM, AMBARI_JSON_COMPONENT_STORM_UI_SERVER);
             return new HostPort(stormUiComp.getHosts().get(0), stormUiComp.getPort());
         }
     }
 
     /**
      * @return List of storm topologies as returned by Storm's REST API
      */
-    public StormTopologies getTopologies() throws IOException {
-        final Map<String, ?> jsonAsMap = JsonClientUtil.getEntity(httpClient.target(tplgySumUrl), Map.class);
-        List<String> topologies = Collections.emptyList();
-        if (jsonAsMap != null) {
-            final List<Map<String, String>> topologiesSummary = (List<Map<String, String>>) jsonAsMap.get(STORM_REST_API_TOPOLOGIES_KEY);
-            if (topologiesSummary != null) {
-                topologies = new ArrayList<>(topologiesSummary.size());
-                for (Map<String, String> tpSum : topologiesSummary) {
-                    topologies.add(tpSum.get(STORM_REST_API_TOPOLOGY_ID_KEY));
+    public StormTopologies getTopologies() throws IOException, PrivilegedActionException {
+        return executeSecure(() -> {
+                    final Map<String, ?> jsonAsMap = JsonClientUtil.getEntity(httpClient.target(tplgySumUrl), Map.class);
+                    List<String> topologies = Collections.emptyList();
+                    if (jsonAsMap != null) {
+                        final List<Map<String, String>> topologiesSummary = 
+                                (List<Map<String, String>>) jsonAsMap.get(STORM_REST_API_TOPOLOGIES_KEY);
+                        if (topologiesSummary != null) {
+                            topologies = new ArrayList<>(topologiesSummary.size());
+                            for (Map<String, String> tpSum : topologiesSummary) {
+                                topologies.add(tpSum.get(STORM_REST_API_TOPOLOGY_ID_KEY));
+                            }
+                        }
+                    }
+                    return new StormTopologies(topologies, getSecurity());
                 }
-            }
-        }
-        return new StormTopologies(topologies, new Security(securityContext, new Authorizer(false), getPrincipals(), getKeytabs()));
+        );
+    }
+
+    private <T, E extends Exception> T executeSecure(SupplierException<T, E> action) throws PrivilegedActionException, E {
+        return action.get();    // For now it executes without security context. Uncomment bellow for security context.
+//        return SecurityUtil.execute(action, securityContext, subject);
```

Comment:
I would say let's just merge it. If it does not work, the fix is to comment the line above and uncomment this line. I will try to find the stack trace. I have to re-run the tests.

Bottom line is, until the discussion that happened yesterday about things stopped working in insecure cluster, all the time things were as they are now (in this patch). So this as is is no different from what we have had the last 2/3 weeks..